File: LensConnect_SampleProgram_Linux_C_x86_x64_1.0.0/LensConnect_Controller/C/LensSetup.c
#include"LensCtrl.h"
#include"LensConnect.h"

void ZoomSpeedChange() {
	int speedPPS = 0;
	uint16_t speedMin = 0;
	uint16_t speedMax = 0;
	ZoomSpeedMinRead(&speedMin);
	ZoomSpeedMaxRead(&speedMax);
	printf("\nPlease input PPS between %d to %d\n", speedMin, speedMax);
	scanf("%u", &speedPPS);
	ZoomSpeedWrite(speedPPS);
}

void FocusSpeedChange() {
	int speedPPS = 0;
	uint16_t speedMin = 0;
	uint16_t speedMax = 0;
	FocusSpeedMinRead(&speedMin);
	FocusSpeedMaxRead(&speedMax);
	printf("\nPleas input PPS between %d to %d\n", speedMin, speedMax);
	scanf("%u", &speedPPS);
	FocusSpeedWrite(speedPPS);
}
void IrisSpeedChange() {
	int speedPPS = 0;
	uint16_t speedMin = 0;
	uint16_t speedMax = 0;
	IrisSpeedMinRead(&speedMin);
	IrisSpeedMaxRead(&speedMax);
	printf("\nPlease input PPS between %d to %d\n", speedMin, speedMax);
	scanf("%u", &speedPPS);
	IrisSpeedWrite(speedPPS);
}
void ZoomBacklashToggleCtl() {
	uint16_t flag = 0;
	printf("\nZoom backlash correction ");
	ZoomBacklashRead(&flag);
	if (flag == ON) {
		ZoomBacklashWrite(OFF);
		printf("OFF\n");
	}
	else {
		ZoomBacklashWrite(ON);
		printf("ON\n");
	}
}
void FocusBacklashToggleCtl() {
	uint16_t flag = 0;
	printf("\nFocus backlash correction ");
	FocusBacklashRead(&flag);
	if (flag == ON) {
		FocusBacklashWrite(OFF);
		printf("OFF\n");
	}
	else {
		FocusBacklashWrite(ON);
		printf("ON\n");
	}
}
void IrisBacklashToggleCtl() {
	uint16_t flag = 0;
	printf("\nIris backlash correction ");
	IrisBacklashRead(&flag);
	if (flag == ON) {
		IrisBacklashWrite(OFF);
		printf("OFF\n");
	}
	else {
		IrisBacklashWrite(ON);
		printf("ON\n");
	}
}

File: LensSetup.py
import LensCtrl
import DefVal as DV

def ZoomSpeedChange():
    retval, speedMin = LensCtrl.ZoomSpeedMinRead()
    retval, speedMax = LensCtrl.ZoomSpeedMaxRead()
    print("\nPlease input PPS between {} to {}" .format(speedMin,speedMax) )
    speedVal = input(">>")
    speedPPS = int(speedVal)
    LensCtrl.ZoomSpeedWrite(speedPPS)

def FocusSpeedChange():
    retval, speedMin = LensCtrl.FocusSpeedMinRead()
    retval, speedMax = LensCtrl.FocusSpeedMaxRead()
    print("\nPlease input PPS between {} to {}" .format(speedMin, speedMax))
    speedVal = input(">>")
    speedPPS = int(speedVal)
    LensCtrl.FocusSpeedWrite(speedPPS)

def IrisSpeedChange():
    retval, speedMin = LensCtrl.IrisSpeedMinRead()
    retval, speedMax = LensCtrl.IrisSpeedMaxRead()
    print("\nPlease input PPS between {} to {}" .format(speedMin, speedMax))
    speedVal = input(">>")
    speedPPS = int(speedVal)
    LensCtrl.IrisSpeedWrite(speedPPS)

def ZoomBacklashToggleCtl():
    print("\nZoom backlash correction ", end="")
    retval, flag = LensCtrl.ZoomBacklashRead()
    if (flag == DV.ON):
        LensCtrl.ZoomBacklashWrite(DV.OFF)
        print("OFF")
    else:
        LensCtrl.ZoomBacklashWrite(DV.ON)
        print("ON")

def FocusBacklashToggleCtl():
    print("\nFocus backlash correction ", end="")
    retval, flag = LensCtrl.FocusBacklashRead()
    if (flag == DV.ON):
        LensCtrl.FocusBacklashWrite(DV.OFF)
        print("OFF")
    else:
        LensCtrl.FocusBacklashWrite(DV.ON)
        print("ON")

def IrisBacklashToggleCtl():
    print("\nIris backlash correction ", end="")
    retval, flag = LensCtrl.IrisBacklashRead()
    if (flag == DV.ON):
        LensCtrl.IrisBacklashWrite(DV.OFF)
        print("OFF")
    else:
        LensCtrl.IrisBacklashWrite(DV.ON)
        print("ON")
